src/Pages/PeoplePage.tsx

```diff
@@ -0,0 +1,219 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { Link, useSearchParams } from 'react-router-dom';
+import { useContext, useEffect } from 'react';
+import classNames from 'classnames';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { PeopleContex } from '../store/PeopleContex';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { getPeople } from '../api';
+import { filterPeople } from '../services/filter';
+import { getSearchWith } from '../utils/searchHelper';
+import { SortType } from '../types/SortType';
+
+export const PeoplePage = () => {
+  const {
+    people,
+    errorMassage,
+    loading,
+    setLoading,
+    setPeople,
+    setErrorMassage,
+  } = useContext(PeopleContex);
+
+  const [searchParams] = useSearchParams();
+
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sex = searchParams.get('sex') || '';
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const filteredPeople =
+    people && filterPeople(people, query, centuries, sex, sort, order);
+  const isPeopleList = people && !!people?.length && !loading;
+  const noPeopleMessage = !filteredPeople?.length && !loading;
+  const isErrorMassage = errorMassage && !loading;
+
+  const handleSorting = (sortType: SortType) => {
+    if (sort === sortType && order === 'desc') {
+      return { sort: null, order: null };
+    }
+
+    if (sort === sortType) {
+      return { sort: sortType, order: 'desc' };
+    }
+
+    return { sort: sortType, order: null };
+  };
+
+  useEffect(() => {
+    setLoading(true);
+
+    getPeople()
+      .then(response => {
+        setPeople(response);
+      })
+      .catch(() => {
+        setErrorMassage(true);
+      })
+      .finally(() => setLoading(false));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+
+      <div className="block">
+        <div className="columns is-desktop is-flex-direction-row-reverse">
+          {isPeopleList && (
+            <div className="column is-7-tablet is-narrow-desktop">
+              <PeopleFilters />
+            </div>
+          )}
+
+          <div className="column">
+            <div className="box table-container">
+              {loading ? (
+                <Loader />
+              ) : (
+                isPeopleList && (
```

Comment:
I like the next syntax:
```
{loading && <Loader />}
{!loading && isPeopleList && (...)}
```
It's more readable for me.
You can try it in your future work if you also like such an approach.